app/models/pacbio/pool.rb

```diff
@@ -46,18 +74,44 @@ def sequencing_runs
 
     private
 
+    def destroy_unused_libraries
+      libraries.filter do |lib|
+        used_aliquots.pluck(:source_id).exclude?(lib.pacbio_request_id)
+      end.each(&:destroy)
+    end
+
+    def destroy_unused_aliquots
+      used_aliquots.filter do |aliquot|
+        libraries.pluck(:pacbio_request_id).exclude?(aliquot.source_id)
+      end.each(&:destroy)
+    end
+
     def update_library(attributes)
       id = attributes['id'].to_s
       indexed_libraries.fetch(id) { missing_library(id) }
                        .tap { |l| l.update(attributes) }
     end
 
+    def update_used_aliquot(attributes)
+      id = attributes['id'].to_s
+      indexed_used_aliquots.fetch(id) { missing_used_aliquot(id) }
+                           .tap { |a| a.update(attributes) }
+    end
+
     def missing_library(id)
-      raise ActiveRecord::RecordNotFound, "Pacbio request #{id} is not part of the pool"
+      raise ActiveRecord::RecordNotFound, "Pacbio library is not part of the pool #{id}"
+    end
+
+    def missing_used_aliquot(id)
+      raise ActiveRecord::RecordNotFound, "Aliquot is not part of the pool #{id}"
     end
 
     def indexed_libraries
```

Comment:
And again. Very similar. Worth looking at abstraction.

Reply:
Yes I was thinking this code is going to be removed soon anyway so I dont want to spend ages on it but Ill admit this is glaringly refactor-able so I will give it a pass over and see what is easily abstracted.